### PR TITLE
Azure Monitor Query Stream Style Serialization Migration

### PR DESCRIPTION
**Pull request by jairmyree**
Migrating azure-monitor-query to stream-style serialization.

### CODE REVIEW

**Comment by jairmyree** [comment]
@alzimmermsft @srnagar There's are two samples in `ReadmeSamples` that reference the usage of Jackson assuming the customer wants to utilize Jackson. Should I replace the sample with a sample using azure-json? Or at least add an additional sample for the case where the customer prefers to use azure-json?

**Comment by srnagar** [comment]
> @alzimmermsft @srnagar There's are two samples in ReadmeSamples that reference the usage of Jackson assuming the customer wants to utilize Jackson. Should I replace the sample with a sample using azure-json? Or at least add an additional sample for the case where the customer prefers to use azure-json?

We should add samples for the case where the customer prefers to use azure-json. I think we can still keep the sample with Jackson but move the Jackson dependency to `test` scope.

File: sdk/monitor/azure-monitor-query/src/main/java/module-info.java [hunk 1]
```diff
@@ -6,12 +6,13 @@
  */
 module com.azure.monitor.query {
     requires transitive com.azure.core;
+    requires com.azure.json;
```
Comment (alzimmermsft) [inline sdk/monitor/azure-monitor-query/src/main/java/module-info.java:9]: This isn't needed anymore as `azure-core` requires this transitively now

File: sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/LogsQueryClientTest.java [hunk 2]
```diff
@@ -352,6 +354,21 @@ query, null, new LogsQueryOptions().setIncludeStatistics(true).setIncludeVisuali
         assertEquals(1, queryResults.getAllTables().size());
         assertNotNull(queryResults.getVisualization());
 
+        BinaryData visualization = queryResults.getVisualization();
+
+        try ( JsonReader reader = JsonProviders.createReader(visualization.toStream());) {
```
Comment (alzimmermsft) [inline sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/LogsQueryClientTest.java:359]:
```suggestion
        try (JsonReader reader = JsonProviders.createReader(visualization.toStream())) {
```

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**